src/components/leaderboards/Paginator.vue

```diff
@@ -0,0 +1,131 @@
+<script setup lang="ts">
+import { computed } from 'vue'
+
+const props = defineProps<{
+	total: number
+	limit: number
+	page: number
+}>()
+
+const limits = [25, 50, 100]
+	.filter((l) => l < props.limit)
+	.concat(props.limit, ...[25, 50, 100].filter((l) => l > props.limit))
```

Comment:
the mention of `64` being a default and it acts as such here, but its `0` do you mean that `props.limit` value is `0`,  and if the behavior is that it'll be `64`, `64` should be shown. Do we need usage of an `onMounted()` hook?

Reply:
We might, yeah. Honestly it's more a question of whether or not we want to fix this graphical bug

Reply:
Calling lifecycle hooks won't work, actually. The value of `limit` passed to `Paginator` takes the `resultsPerPage` query param if it exists, defaulting to the caller component's `limitDefault` otherwise. And `limitDefault` is defined as 0 until the API call's actually made. I'll just prevent rendering `Paginator` when its `limit` prop is zero.

Reply:
I've fixed it, though it's pretty strange. Tried adding `v-if="props.limit > 0"` to `Paginator`'s template, but that doesn't fix things. The `limit` prop won't update along with an update to the parent's `limitDefault`. What does fix is where `Paginator`'s called in the parent. In order to prevent this bug, we have to call `Paginator` in a block where the API call has already been made.